crates/proof-of-sql-parser/src/sql.lalrpop

```diff
@@ -435,4 +441,6 @@ match {
     // Integer numbers (without a fractional part)
     r"[+-]?[0-9]+" => INTEGER_LIT,
     r"'(?s)(?:''|[^'])*'" => STRING_LITERAL,
+    // Timestamp literals with optional fractional seconds and time zone offset
+    r"\d{4}-\d{2}-\d{2} \d{2}:\d{2}:\d{2}(?:\.\d{1,9})?(?:Z|[+-]\d{2}:\d{2})?" => TIMESTAMP_LIT,
```

Comment:
We had discussed a method of just parsing out strings from the lexer instead of timestamps and I plan to attempt that approach again, I was receiving a "ambiguous grammar" error when trying to add literal support, so I am simply doing this for now for testing purposes so I can come back to this later